pkg/agent/wireguard/client_linux.go

```diff
@@ -145,18 +150,34 @@ func (client *client) Init() error {
 		ListenPort:   &client.wireGuardConfig.Port,
 		ReplacePeers: false,
 	}
-	patch, _ := json.Marshal(map[string]interface{}{
-		"metadata": map[string]interface{}{
-			"annotations": map[string]string{
-				types.NodeWireGuardPublicAnnotationKey: client.privateKey.PublicKey().String(),
+	if !multiclusterEnabled {
+		patch, _ := json.Marshal(map[string]interface{}{
+			"metadata": map[string]interface{}{
+				"annotations": map[string]string{
+					types.NodeWireGuardPublicAnnotationKey: client.privateKey.PublicKey().String(),
+				},
 			},
-		},
-	})
-	if err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
-		_, err := client.k8sClient.CoreV1().Nodes().Patch(context.TODO(), client.nodeName, apitypes.MergePatchType, patch, metav1.PatchOptions{}, "status")
-		return err
-	}); err != nil {
-		return fmt.Errorf("error when patching the Node with the '%s' annotation: %w", types.NodeWireGuardPublicAnnotationKey, err)
+		})
+		if err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+			_, err := client.k8sClient.CoreV1().Nodes().Patch(context.TODO(), client.nodeName, apitypes.MergePatchType, patch, metav1.PatchOptions{}, "status")
+			return err
+		}); err != nil {
+			return fmt.Errorf("error when patching the Node with the '%s' annotation: %w", types.NodeWireGuardPublicAnnotationKey, err)
+		}
+	} else {
+		patch, _ := json.Marshal(map[string]interface{}{
+			"wireGuard": map[string]interface{}{
+				"publicKey": client.privateKey.PublicKey().String(),
+				"port":      client.wireGuardConfig.Port,
+			},
+		})
+		if err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+			_, err := client.mcClient.MulticlusterV1alpha1().Gateways(client.mcNamespace).Patch(context.TODO(), client.nodeName, apitypes.MergePatchType, patch,
```

Comment:
I feel better to encapsulate the MC Gateway lookup logic in a multi-cluster package, rather than wireguard code. Could we get the `Gateway` resource in the caller and pass it to `Init()` instead? In this way, we do not even need the `multiclusterEnabled` argument anymore.

Reply:
Agree. The public key will be returned in `Init()` now, so that we can lookup MC Gateway in `mc_route_controller.go`. And the WireGuard package won't  be coupled with Multi-cluster package now.